File: Examples/test-suite/php/director_thread_runme.php
<?php

require "tests.php";

# Fails in a ZTS-build of PHP5 - see: https://github.com/swig/swig/pull/155
# FIXME: Does this still fail in a threaded build of PHP7?
exit(0);

// New functions
check::functions(array('millisecondsleep','foo_stop','foo_run','foo_do_foo'));
// New classes
check::classes(array('director_thread','Foo'));
// No new vars
check::globals(array());

class Derived extends Foo {
  function do_foo() {
    $this->val = $this->val - 1;
  }
}

$d = new Derived();
$d->run();

if ($d->val >= 0) {
  check::fail($d->val);
}

$d->stop();

check::done();

File: Examples/test-suite/lua/director_thread_runme.lua
require("import") -- the import fn
import("director_thread") -- import lib

local d = director_thread.Foo()
function do_foo(self)
   self.val = self.val - 1
end
swig_override(d, 'do_foo', do_foo)
d:run()
assert(d.val < 0)
d:stop()
